editor/hooks/autocomplete/index.js

```diff
@@ -0,0 +1,56 @@
+/**
+ * External dependencies
+ */
+import { compose } from 'lodash';
+import { connect } from 'react-redux';
+
+/**
+ * WordPress dependencies
+ */
+import { Autocomplete, withContext } from '@wordpress/components';
+import { addFilter } from '@wordpress/hooks';
+
+/**
+ * Internal dependencies
+ */
+import { blockAutocompleter, userAutocompleter } from './autocompleters';
+import { getInserterItems } from '../../store/selectors';
+
+function withAutocomplete( BlockEdit ) {
```

Comment:
I really like this approach using a HOC.

Reply:
Agreed 😃